uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-fs/src/main/java/org/uberfire/java/nio/fs/file/SimpleFileSystemProvider.java

```diff
@@ -744,6 +622,112 @@ private GeneralPathImpl toGeneralPathImpl(final Path path) {
         return s;
     }
 
+    protected FileSystem getDefaultFileSystem() {
+        return fileSystem;
+    }
+
+    protected void checkFileNotExistThenThrow(final Path path, boolean isLink) {
+        if (!path.toFile().exists()) {
+            if (isLink) {
+                throw new NotLinkException(path.toString());
+            } else {
+                throw new NoSuchFileException(path.toString());
+            }
+        }
+    }
+
+    protected void checkFileExistsThenThrow(final Path path) {
+        if (path.toFile().exists()) {
+            throw new FileAlreadyExistsException(path.toString());
+        }
+    }
+
+    private final class SimpleFileSystemDirectoryStream implements DirectoryStream<Path> {
+
+        private final class SimpleFileSystemIterator implements Iterator<Path> {
+
+            public boolean atEof = false;
+            private int i = -1;
+            private Path nextEntry = null;
+
+            @Override
+            public boolean hasNext() {
+                if (nextEntry == null && !atEof) {
+                    nextEntry = readNextEntry();
+                }
+                return nextEntry != null;
+            }
+
+            @Override
+            public Path next() {
+                final Path result;
+                if (nextEntry == null && !atEof) {
+                    result = readNextEntry();
+                } else {
+                    result = nextEntry;
+                    nextEntry = null;
+                }
+                if (result == null) {
+                    throw new NoSuchElementException();
+                }
+                return result;
+            }
+
+            private Path readNextEntry() {
+                if (atEof) {
+                    return null;
+                }
+
+                Path result = null;
+                while (true) {
+                    i++;
+                    if (i >= content.length) {
+                        atEof = true;
+                        break;
+                    }
+
+                    final Path path = content[i];
+                    if (filter.accept(path)) {
+                        result = path;
+                        break;
+                    }
+                }
+
+                return result;
+            }
+
+            @Override
+            public void remove() {
+                throw new UnsupportedOperationException();
+            }
+        }
+
+        private final Filter<Path> filter;
+        private final Path[] content;
+        boolean isClosed = false;
+
+        private SimpleFileSystemDirectoryStream(Filter<Path> filter, Path[] content) {
+            this.filter = filter;
+            this.content = content;
+        }
+
+        @Override
+        public void close() throws IOException {
+            if (isClosed) {
```

Comment:
I think we should do the same check also in `next()` and `hasNext()` methods

Reply:
changes as per suggested